api/query/cache/service/main.go

```diff
@@ -118,8 +118,15 @@ func searchHandler(w http.ResponseWriter, r *http.Request) {
 
 	// Return to client `http.StatusUnprocessableEntity`immediately if any runs
 	// are missing.
-	if len(missing) != 0 {
-		http.Error(w, fmt.Sprintf("Test run(s) with ID(s) %v not loaded into cache. Try again later.", missing), http.StatusUnprocessableEntity)
+	if len(runs) == 0 {
```

Comment:
Might be redundant based on earlier code, but check that `len(missing) > 0` too.